valhalla/macro.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <exception>
+
+#define CHECK_THROWS(condition, message)                                                             \
+  {                                                                                                  \
+    if (!(condition))                                                                                \
+      throw std::invalid_argument(message);                                                          \
```

Comment:
does something catch this exception further up and convert it into a valhalla_exception_t ? because those are the only ones that get turned into proper messages that the user of the service will see.

Reply:
good question. I think it's out of responsibility of this code. `std::invalid_argument` is inherited from `std::exception`, so if we handle `std::exception` everything should be ok